zzz_generated/all-clusters-app/zap-generated/endpoint_config.h

```diff
@@ -595,20 +595,26 @@
                                                                                                                                    \
             /* Endpoint: 1, Cluster: Pump Configuration and Control (server), big-endian */                                        \
                                                                                                                                    \
-            /* 4574 - LifetimeEnergyConsumed, */                                                                                   \
+            /* 4574 - LifetimeRunningHours, */                                                                                     \
+            0x00, 0x00, 0x00,                                                                                                      \
```

Comment:
So the good news is that the default is in fact 3 bytes.  If the default were a value where we could see the byte ordering, that would be helpful in terms of seeing whether reading as always-little-endian makes sense.

Reply:
I'll try to make a test with other default values.

Reply:
There are two DEFAULT tables in the `endpoint_config.h` fiel. One for big-endian CPU's and one for little-endian CPU's. So I think this is taken care of when compiling. So the `typeSensitiveMemCopy()` will work correctly, since all data are stored locally on the device according to it's endianess - at least that is what I can conclude from the code.

Reply:
But the zap tool does not generate it correctly. I can see the following in the `endpoint_config.h` file after setting the `LifetimeRunningHours` default value to 0x123456 in the zap tool UI and regenrating the files using `./scripts/tools/zap_regen_all.py`.

```c
            /* Endpoint: 1, Cluster: Pump Configuration and Control (server), big-endian */                                        \
                                                                                                                                   \
            /* 1377 - LifetimeRunningHours, */                                                                                     \
            0x12, 0x34, 0x56,                                                                                                      \
```

and

```c
            /* Endpoint: 1, Cluster: Pump Configuration and Control (server), little-endian */                                     \
                                                                                                                                   \
            /* 1377 - LifetimeRunningHours, */                                                                                     \
            0x12, 0x34, 0x56,                                                                                                      \
```

This is not expected, since these are encapsulated in an if-else:

```c
// Default values for the attributes longer than a pointer,
// in a form of a binary blob
// Separate block is generated for big-endian and little-endian cases.
#if BIGENDIAN_CPU
#define GENERATED_DEFAULTS                                                                                                         \
    {         
......
    }
#else // !BIGENDIAN_CPU
#define GENERATED_DEFAULTS                                                                                                         \
    {                                                                                                                              \
......
    }
#endif // BIGENDIAN_CPU
``

Reply:
Should be fixed now, with https://github.com/project-chip/connectedhomeip/pull/11451 merged.